spec/models/update_subscriber_list_spec.rb

```diff
@@ -0,0 +1,119 @@
+require "spec_helper"
+
+RSpec.describe UpdateSubscriberList do
+  let(:content_id) { SecureRandom.uuid }
+  let(:document_title) { "Example Title" }
+  let(:major_change_document) do
+    {
+      "base_path" => "/foo",
+      "content_id" => content_id,
+      "title" => document_title,
+      "update_type" => "major",
+    }
+  end
+
+  let(:minor_change_document) do
+    {
+      "base_path" => "/foo",
+      "content_id" => content_id,
+      "title" => document_title,
+      "update_type" => "minor",
+    }
+  end
+
+  let(:logger) { double(:logger, info: nil) }
+  let(:update_subscriber_list) { UpdateSubscriberList.new(document, logger) }
+  let(:subscriber_list_slug) { "subscriber_list_slug" }
+  let(:updateable_parameters) { { "title" => document_title } }
+
+  let(:subscriber_list_attributes) do
+    {
+      "content_id" => content_id,
+      "slug" => subscriber_list_slug,
+      "title" => subscriber_list_title,
+    }
+  end
+
+  describe "#trigger" do
+    shared_examples "an attempt to update subscriber lists with appropriate logging" do
+      let(:subscriber_list_title) { "An old outdated title" }
+
+      context "subscriber list found by content id" do
+        before { stub_email_alert_api_has_subscriber_list(subscriber_list_attributes) }
+
+        it "logs an attempt to update the subscriber list and triggers the api to update the title" do
+          stub_email_alert_api_has_subscriber_list(subscriber_list_attributes)
+          stub_update_subscriber_list_details(slug: subscriber_list_slug, params: updateable_parameters)
+          update_subscriber_list.trigger
+
+          expect(logger).to have_received(:info).with(
+            "Attempting to update subscriber list: #{subscriber_list_slug}, with: #{updateable_parameters}",
+          )
+        end
+
+        context "subscriber list title matches document list" do
+          let(:subscriber_list_title) { document_title }
+
+          it "logs the outcome and does not attempt an update request" do
+            stub_update_subscriber_list_details(slug: subscriber_list_slug, params: updateable_parameters)
+            update_subscriber_list.trigger
+
+            expect(logger).to have_received(:info).with(
+              "No update needed to subscriber list: #{subscriber_list_slug}",
+            )
+          end
+        end
+
+        context "update attempted with invalid parameters" do
+          let(:document_title) { "" }
```

Comment:
This change isn't actually important, right?  No behaviour in email-alert-service changes if the document title is the empty string

Reply:
No that's fair - it's just a different scenario the test helpers can cater for.
You get a slightly differen toutput from the logger as `updateable_parameters` is `{ title => ""}` instead of `{}`

But that's all this tests.

Reply:
@barrucadu Given the above, do you still want it removed?

Reply:
I think remove it, as otherwise it suggests that `document_title` being `""` is important here.